### DISCUSSION TITLE
Resolution and scaling of KUDs

### DISCUSSION THREAD

**Discussion by ecbolas** [jmsigner/amt#99] [Q&A] [answered]
Hello, 

I have a couple questions about scaling and resolution->

I am making individual KUDs, that I want to scale between 0-1, and then average into a single UD to represent the population. The reason for scaling is to account for different sample sizes between individuals, but I want each of their UD’s to have the same “weight” when averaged into the population-level UD. This population UD raster will then be used as a covariate in an RSF. The other covariates/rasters for the RSF are 30m resolution, so I want to make sure my population UD matches that resolution.

My questions are:
1)	When I make the template raster for the KUDs and give it a resolution, what is the unit of resolution? In other words, what does the number mean that I am giving it?

2)	For my work flow, I am using hr_kde, then hr_ud. However, I noticed that hr_kde also returns a UD, and that the max and min values for the two UD calculations are different, can you explain that please? For example, the hr_kde UD max value goes to any number, but the hr_ud UD min/max returns numbers between 0-1 (see below with outputs of the two different UDs for the same animal). Which is related to my next question -> 

3)	If all the individual KUDs were made with the same template raster, are they already on the same scale/scaled to each other? Is the UD function already scaling them? I’m trying to determine if I need to use terra::rescale before I average them together, or if they are already on an equivalent scale. When I apply terra::rescale, the max and min values come back the same. However, while the plot of the hr_KDE UD and hr_ud look identical to one another, the rescaled UD plot looks different.

Code:
# make kdes
catsHR<- catstrack1 %>% 
  mutate(HR_KDE = map(data, ~hr_kde(., trast = base_trast, levels = c(0.95, 1))))
# make UDs
catsHR2<- catsHR %>% 
  mutate(UD = map(HR_KDE, hr_ud))

# look at UD in hr_kde vs UD from kr_ud for the same individual
> catsHR$HR_KDE[[1]]$ud
class       : SpatRaster 
dimensions  : 134, 200, 1  (nrow, ncol, nlyr)
resolution  : 0.0062052, 0.0062052  (x, y)
extent      : -119.4016, -118.1605, 33.70379, 34.53528  (xmin, xmax, ymin, ymax)
coord. ref. : lon/lat WGS 84 (EPSG:4326) 
source(s)   : memory
name        :    lyr.1 
min value   :    0.000 
max value   : 1392.781 

> catsHR2$UD[[1]]
class       : SpatRaster 
dimensions  : 134, 200, 1  (nrow, ncol, nlyr)
resolution  : 0.0062052, 0.0062052  (x, y)
extent      : -119.4016, -118.1605, 33.70379, 34.53528  (xmin, xmax, ymin, ymax)
coord. ref. : lon/lat WGS 84 (EPSG:4326) 
source(s)   : memory
name        :      lyr.1 
min value   : 0.00000000 
max value   : 0.05430307 

Thank you so much! Feel free to comment on anything that I’ve proposed doing and I can share more code if needed 😊

**Comment by jmsigner** [comment, marked as answer]
Hi, 

below some answers: 

> 1. When I make the template raster for the KUDs and give it a resolution, what is the unit of resolution? In other words, what does the number mean that I am giving it?

The units are the units. In you case most likely meters.

> 2. For my work flow, I am using hr_kde, then hr_ud. However, I noticed that hr_kde also returns a UD, and that the max and min values for the two UD calculations are different, can you explain that please? For example, the hr_kde UD max value goes to any number, but the hr_ud UD min/max returns numbers between 0-1 (see below with outputs of the two different UDs for the same animal)

`hr_ud()`returns a UD that sums to 1, i.e., a proper pdf. You can verify this by

```
library(amt)

data("deer")
kde1 <- hr_kde(deer)

kde1$ud
hr_ud(kde1)

# You can get the same by manually standardizing
kde1$ud / sum(kde1$ud[], na.rm = TRUE)
```

> 3. If all the individual KUDs were made with the same template raster, are they already on the same scale/scaled to each other? Is the UD function already scaling them?

I am not sure that I fully understand this. They are on the same resolution and if you use `hr_ud()`they will also sum to one. So yes, in some sense they are on the same scale. Does this help?

**Reply by ecbolas** [reply]
Hi thanks so much for the reply!

Your example for question two is super helpful clarifying the scaling that hr_ud does, it's great that the function is already scaling for me.

I've come across one additional follow-up question that is related to the above: For the template raster, should this be created with the data set for all animals, or should I be making each individual it's own trast? I want output resolutions for my KDEs/UDs of 30m, but something is going wrong with how I'm specifying when I use the dataset of the entire population, since the output resolution is huge! Code below:


# track of entire population
catstrack<- amt::make_track(samocats_dfutm, .x = utmE, .y = utmN, .t = DATE_TIME, id = ID, crs = 32611)


# nest data to turn it into list by individual
catstrack1<- catstrack %>% 
  nest(data = -c(id))

base_trast<- make_trast(catstrack, res = 30)
class       : SpatRaster 
dimensions  : 138, 200, 1  (nrow, ncol, nlyr)
resolution  : 743.2913, 743.2913  (x, y)
extent      : 269944.8, 418603, 3728030, 3830605  (xmin, xmax, ymin, ymax)
coord. ref. : WGS 84 / UTM zone 11N (EPSG:32611)

**Reply by ecbolas** [reply]
I can also run hr_kde without a template raster, which returns individual UDs with more reasonable resolutions (56 x56), but ultimately I still need each UD to be have the same resolution so that I can average them together.

**Reply by jmsigner** [reply]
I would create the template raster for the extent of all animals. This will make it easier to average it later. In regions where an animal does not have any data, the UD will be essentially 0, so nothing to worry about.
